flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/runtime/utils/StreamingWithStateTestBase.scala

```diff
@@ -68,8 +69,8 @@ class StreamingWithStateTestBase(state: StateBackendMode) extends StreamingTestB
     super.before()
     // set state backend
 
-    // subfolder are managed here because the tests could fail during cleanup when concurrently executed (see FLINK-33820)
-    baseCheckpointPath = TempDirUtils.newFolder(tempFolder)
+    val baseCheckpointPath = Files.createTempDirectory(getClass.getCanonicalName)
+    Files.deleteIfExists(baseCheckpointPath);
```

Comment:
>baseCheckpointPath = Files.createTempDirectory("junit")

I didn't get why can't we use `baseCheckpointPath = Files.createTempDirectory(getClass.getCanonicalName)` ?
Havig classname in prefix could simplify finding root cause in case of issues IMHO

The idea is to use `Files.createTempDirectory` and `deleteIfExists` to be sure that delete is happening at the end and it should not intersect with checkpoint removal

Reply:
It's ok to use `Files.createTempDirectory(getClass.getCanonicalName)`.
`TempDirUtils.newFolder(tempFolder)` may still cause problems, it's my fault of PR 33641.